src/foam/dao/index/Plan.js

```diff
@@ -166,3 +166,79 @@ foam.CLASS({
     }
   ]
 });
+
+/**
+  Merges results from multiple sub-plans and deduplicates, sorts, and
+  filters the results.
+
+  TODO: account for result sorting in cost?
+*/
+foam.CLASS({
+  package: 'foam.dao.index',
+  name: 'MergePlan',
+  extends: 'foam.dao.index.AltPlan',
+
+  requires: [
+    'foam.dao.DedupSink',
+    'foam.dao.LimitedSink',
+    'foam.dao.SkipSink',
+    'foam.dao.OrderedSink',
+  ],
+
+  methods: [
+    /**
+      Executes sub-plans, limiting results from each, then merges results,
+      removes duplicates, sorts, skips, and limits.
+    */
+    function execute(promise, sink, skip, limit, order, predicate) {
+      // TODO: Investigate pre-sorted results from subqueries being
+      //   zipped together quickly
+
+      var resultSink = this.DedupSink.create({
+        delegate: this.decorateSink_(sink, skip, limit, order)
```

Comment:
Yes, the TODO indicates this improvement should be implemented. I'll add an issue for it